src/PowerShellEditorServices/Services/PowerShell/Refactoring/IterativeVariableVisitor.cs

```diff
@@ -0,0 +1,475 @@
+                    {
+                        Splatted = varAst;
+                        break;
+                    }
+                }
+                if (Splatted != null)
+                {
+                    NewSplattedModification(Splatted);
+                }
+                else
+                {
+                    // The Target Variable is a Parameter and the commandAst is the Target Function
+                    ShouldRename = true;
+                }
+            }
+        }
+
+        private void ProcessVariableExpressionAst(VariableExpressionAst variableExpressionAst)
+        {
+            if (variableExpressionAst.VariablePath.UserPath.ToLower() == OldName.ToLower())
+            {
+                // Is this the Target Variable
+                if (variableExpressionAst.Extent.StartColumnNumber == StartColumnNumber &&
+                variableExpressionAst.Extent.StartLineNumber == StartLineNumber)
+                {
+                    ShouldRename = true;
+                    TargetVariableAst = variableExpressionAst;
+                }
+                // Is this a Command Ast within scope
+                else if (variableExpressionAst.Parent is CommandAst commandAst)
+                {
+                    if (WithinTargetsScope(TargetVariableAst, commandAst))
+                    {
+                        ShouldRename = true;
+                    }
+                }
+                // Is this a Variable Assignment thats not within scope
+                else if (variableExpressionAst.Parent is AssignmentStatementAst assignment &&
+                    assignment.Operator == TokenKind.Equals)
+                {
+                    if (!WithinTargetsScope(TargetVariableAst, variableExpressionAst))
+                    {
+                        ShouldRename = false;
+                    }
+
+                }
+                // Else is the variable within scope
+                else
+                {
+                    ShouldRename = WithinTargetsScope(TargetVariableAst, variableExpressionAst);
+                }
+                if (ShouldRename)
+                {
+                    // have some modifications to account for the dollar sign prefix powershell uses for variables
+                    TextChange Change = new()
+                    {
+                        NewText = NewName.Contains("$") ? NewName : "$" + NewName,
+                        StartLine = variableExpressionAst.Extent.StartLineNumber - 1,
+                        StartColumn = variableExpressionAst.Extent.StartColumnNumber - 1,
+                        EndLine = variableExpressionAst.Extent.StartLineNumber - 1,
+                        EndColumn = variableExpressionAst.Extent.StartColumnNumber + OldName.Length,
+                    };
+                    // If the variables parent is a parameterAst Add a modification
+                    if (variableExpressionAst.Parent is ParameterAst paramAst && !AliasSet)
```

Comment:
I love the Alias auto-append and think it should be default, but we may want to surface a setting to make it optional, I could see this annoying a lot of people.

Reply:
Agreed I've been testing this code for a while personally and I have been wanting the ability to toggle it

Reply:
I've Added a VScode Setting for this shouldGenerateAlias which is now passed to the VariableVisitor class and respected